File: content/blog/number_of_1_bits/go/pop_count.go
package number_of_1_bits

func hammingWeightPopCount(n int) int {
	mask := 1
	count := 0

	for i := 0; i < 32; i++ {
		if n&mask != 0 {
			count++
		}
		mask <<= 1
	}

	return count
}

File: content/blog/number_of_1_bits/ts/pop_count.ts
export const hammingWeightPopCount = (n: number): number => {
    let mask = 1
    let count = 0

    for (let i = 0; i < 32; i++) {
        if (n & mask) {
            count++
        }
        mask <<= 1
    }

    return count
};